terraform/eval_validate.go

```diff
@@ -101,6 +121,64 @@ func (n *EvalValidateProvisioner) Eval(ctx EvalContext) (interface{}, error) {
 	}
 }
 
+func (n *EvalValidateProvisioner) validateConnConfig(connConfig *ResourceConfig) (warns []string, errs []error) {
```

Comment:
Can we move this functionality into the communicator package, or at least somehow get the `connConfigSuperset` definition in there? It wold be nice to have some logical locality next time we're trying to figure out why a new communicator fails validation ;)